packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -129,22 +129,30 @@ const App = (props) => {
     // Get the current currency to be used through out the app
     const currency = locale.preferredCurrency || l10n.defaultCurrency
 
-    // Handle creating a new pasket if there isn't one already assigned to the current
+    // Handle creating a new basket if there isn't one already assigned to the current
     // customer.
     const {data: customer} = useCurrentCustomer()
     const {data: baskets} = useCustomerBaskets(
         {parameters: {customerId: customer.customerId}},
         {enabled: !!customer.customerId && !isServer}
     )
     const createBasket = useShopperBasketsMutation('createBasket')
+    const updateBasket = useShopperBasketsMutation('updateBasket')
 
-    // Create a new basket if the current customer doesn't have one.
     useEffect(() => {
+        // Create a new basket if the current customer doesn't have one.
         if (baskets?.total === 0) {
             createBasket.mutate({
                 body: {}
             })
         }
+        // update the basket currency if it doesn't match the current locale currency
+        if (baskets?.baskets?.[0]?.currency && baskets?.baskets?.[0]?.currency !== currency) {
+            updateBasket.mutate({
```

Comment:
The `udpateBasket` call was required when using [OCAPI createBasket](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/105#:~:text=Cart/Checkout/Add%20to%20cart) endpoint since we couldn't set the currency when creating the basket.

Did we confirm that we need an extra call using SCAPI createBasket? The endpoint accepts a `Basket` with a defined currency. But it may need the case that we need the update call if the basket was created with a different currency originally.

I tested the change locally and still see some weird behaviour with the locale on the basket page.

**STR**
1. Add a product to the cart.
2. Switch to it-IT locale.
3. Switch to es-MX locale http://localhost:3000/global/es-MX/cart 
4. See the cart value 0 and the product's title in the cart is in Italian, not Spanish.

<img width="1843" alt="locale" src="https://user-images.githubusercontent.com/6024102/230140979-7c3c1329-2656-4917-a191-18ea28c06761.png">

Reply:
@alexvuong I still can reproduce the same problem switching to es-MX.